nwb_conversion_tools/tools/spikeinterface/spikeinterface.py

```diff
@@ -400,17 +402,31 @@ def add_electrodes(
                         )
                     electrode_kwargs.update(dict(group=nwbfile.electrode_groups[group_name], group_name=group_name))
                 elif "data" in desc:
-                    electrode_kwargs[name] = desc["data"][j]
+                    electrode_kwargs[name] = desc["data"][data_index]
 
             if "group_name" not in elec_columns:
-                group_id = checked_recording.get_channel_groups(channel_ids=[channel_id])[0]
+                group_id = checked_recording.get_channel_groups(channel_ids=[data_index])[0]
                 electrode_kwargs.update(dict(group=nwbfile.electrode_groups[str(group_id)], group_name=str(group_id)))
 
             nwbfile.add_electrode(**electrode_kwargs)
 
     # add columns for existing electrodes:
+    channels_not_available_in_recorder = set(channel_names_in_electrodes_table).difference(set(channel_name_array))
     for col_name, cols_args in elec_columns_append.items():
+        data = cols_args["data"]
+        samp_data = data[0]
+        data_type_found = [proptype for proptype in property_default_types if isinstance(samp_data, proptype)][0]
+
+        default_value = property_default_types[data_type_found]
```

Comment:
```suggestion
        for proptype in property_default_types:
            if isinstance(data[0], proptype):
                default_value = property_default_types[proptype]
                break
```
Good opportunity here to do what I suggested in previous PR.

Reply:
I still don't feel completely comfortable with this because of the number of for loops. I am trying to reduce https://en.wikipedia.org/wiki/Cyclomatic_complexity of this function as it really bogged me down the first time I attempted to understand it and I want this to be easier for a future developer. 

Check the accepeted answer here that I will implement in a refactor:
https://stackoverflow.com/questions/9868653/find-first-sequence-item-that-matches-a-criterion
If you still don't like it we go with the loop:

Reply:
Great job, a generator approach is even better than what I suggested!